server/lib/availabilitySync.ts

```diff
@@ -57,141 +54,93 @@ class AvailabilitySync {
-            if (
-              !mediaExists &&
-              (season.status !== MediaStatus.UNKNOWN ||
-                season.status4k !== MediaStatus.UNKNOWN)
-            ) {
-              await seasonRepository.update(
-                { id: season.id },
-                {
-                  status: MediaStatus.UNKNOWN,
-                  status4k: MediaStatus.UNKNOWN,
-                }
-              );
-            } else {
-              const seasonExists = await this.seasonExists(media, season);
-
-              if (!seasonExists) {
-                logger.info(
-                  `Removing season ${season.seasonNumber}, media ID ${media.id} because it does not exist in any of your media instances.`,
-                  { label: 'AvailabilitySync' }
-                );
-
-                if (
-                  season.status !== MediaStatus.UNKNOWN ||
-                  season.status4k !== MediaStatus.UNKNOWN
-                ) {
-                  await seasonRepository.update(
-                    { id: season.id },
-                    {
-                      status: MediaStatus.UNKNOWN,
-                      status4k: MediaStatus.UNKNOWN,
-                    }
-                  );
-                }
-
-                const seasonToBeDeleted = await seasonRequestRepository.findOne(
-                  {
-                    relations: {
-                      request: {
-                        media: true,
-                      },
-                    },
-                    where: {
-                      request: {
-                        media: {
-                          id: media.id,
-                        },
-                      },
-                      seasonNumber: season.seasonNumber,
-                    },
-                  }
-                );
-
-                if (seasonToBeDeleted) {
-                  await seasonRequestRepository.remove(seasonToBeDeleted);
-                }
-
-                didDeleteSeasons = true;
-              }
-            }
+          await mediaRepository.save({ media, ...media });
 
-            if (didDeleteSeasons) {
-              if (
-                media.status === MediaStatus.AVAILABLE ||
-                media.status4k === MediaStatus.AVAILABLE
-              ) {
-                logger.info(
-                  `Marking media ID ${media.id} as PARTIALLY_AVAILABLE because season removal has occurred.`,
-                  { label: 'AvailabilitySync' }
-                );
-
-                if (media.status === MediaStatus.AVAILABLE) {
-                  await mediaRepository.update(media.id, {
-                    status: MediaStatus.PARTIALLY_AVAILABLE,
-                  });
-                }
-
-                if (media.status4k === MediaStatus.AVAILABLE) {
-                  await mediaRepository.update(media.id, {
-                    status4k: MediaStatus.PARTIALLY_AVAILABLE,
-                  });
-                }
-              }
-            }
```

Comment:
This has also been moved along with the season checker as we still need to check for season removal to change the media status to partially available.